src/olympia/lib/git.py

```diff
@@ -402,46 +403,41 @@ def iter_tree(self, tree):
                     tree_entry=tree_entry,
                     path=tree_entry.name)
 
-    def get_diff(self, commit, parent=None, pathspec=None):
-        """Get a diff from `parent` to `commit`.
-
-        If `parent` is not given we assume it's the first commit and handle
-        it accordingly.
-
-        We are resolving renames and copies according to a 50% similarity
-        threshold.
+    @lru_cache(maxsize=None)
```

Comment:
`maxsize=None` ? This seems quite dangerous, the cache will grow forever in the python process, which can itself live for quite a while...

Reply:
I think what you need here is something that caches on the object instance - `self` - so that it goes away after you're done with that instance. Just doing `lru_cache(maxsize=1)` might work but it's per-process so it may be it's a little risky. A quick search suggest decorating with `lru_cache` in the object's `__init__`, since the cache is attached to the function being decorated, doing it at `__init__` time might be safe (still need `maxsize=1` though).

Will need some tests in any case.

Reply:
Solved this a bit differently in fb60d6d, I hope that works too? Added a test as well.

Reply:
I figured that would be easier to implement / test compared to re-using lru_cache here too. Especially since lru_cache(maxsize=1) wouldn't give us any benefits of the actual data structure and with the destruction of `self` happening right after serialization we should be safe.